features/steps/utils/geometry.py

```diff
@@ -77,3 +85,41 @@ def is_closed(context, instance):
     precision = get_precision_from_contexts(entity_contexts)
     points_coordinates = get_points(instance)
     return math.dist(points_coordinates[0], points_coordinates[-1]) < precision
+
+
+def evaluate_segment(segment: ifcopenshell.entity_instance, dist_along: float) -> np.ndarray:
+    s = ifcos_geom.settings()
+    pwf = wrapper.map_shape(s, segment.wrapped_data)
+
+    prev_trans_matrix = pwf.evaluate(dist_along)
```

Comment:
How sure are we that this does not do the "interpolation" / "smoothing" with next segment. (The one that messes up multi-threading.)

Do we have some exact hand calculated solutions to verify?

Reply:
There is no "interpolation" / "smoothing" between segments.  That was discussed early in the C++ implementation but it turned out to be difficult and problematic, therefore was dropped.